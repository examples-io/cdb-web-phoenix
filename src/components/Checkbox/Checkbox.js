import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { getClassNames } from '../../utlities/getClassNames'

/**
 * @description Checkbox components - renders a checkbox with a label.
 *              Optional error and disabled styling applied as specified by the props.
 */
class Checkbox extends Component {
    render() {
        return (
            <label
                className={'form__input form__input--inline form__input--checkbox ' + (this.props.className ? this.props.className : '')}
                htmlFor={this.props.id}>
                <input
                    type="checkbox"
                    className={getClassNames('input input--checkbox', 'input--error', this.props.error)}
                    id={this.props.id}
                    value={this.props.value}
                    checked={this.props.value ? 'checked' : false}
                    onChange={this.props.onChange}
                    disabled={this.props.disabled} />
                <span className="checkbox"></span>
                <div className={getClassNames('label label--inline label--checkbox', 'label--error', this.props.error)}>
                    {this.props.label}
                </div>
            </label>
        );
    }
}

Checkbox.propTypes = {
    /**
     * id - element id of the checkbox
     */
    id: PropTypes.string,
    /**
     * label - text to be displayed as the label
     */
    label: PropTypes.string,
    /**
     * value - internal value property of the checkbox
     */
    value: PropTypes.oneOfType([ PropTypes.string, PropTypes.bool, PropTypes.number ]),
    /**
     * onChange - function to be attached to the checkbox's onChange event
     */
    onChange: PropTypes.func,
    /**
     * error - boolean to indicate whether error styling should be applied or not
     */
    error: PropTypes.bool,
    /**
     * disabled - boolean to indicate whether checkbox should be disabled and styling applied or not
     */
    disabled: PropTypes.bool,
    /**
     * className - classes to be applied to the checkbox element
     */
    className: PropTypes.string,
};

/**
 * @description default values for disabled and error prop types
 */
Checkbox.defaultProps = {
    disabled: false,
    error: false
};

export default Checkbox;

