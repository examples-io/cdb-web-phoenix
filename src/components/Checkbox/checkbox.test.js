import React from 'react';
import renderer from 'react-test-renderer';

import Checkbox from './Checkbox';

describe('Checkbox tests', () => {
    it ('renders Checkbox correctly', () => {
        const tree = renderer
            .create(<Checkbox id="testCheckbox" label="Test Checkbox" value="" />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it ('renders Checkbox error correctly', () => {
        const tree = renderer
            .create(<Checkbox id="errorCheckbox" label="Error Checkbox" value="" error />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it ('renders disabled Checkbox correctly', () => {
        const tree = renderer
            .create(<Checkbox id="disabledCheckbox" label="Disabled Checkbox" value="" disabled />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it ('renders Checkbox with custom class correctly', () => {
        const tree = renderer
            .create(<Checkbox id="customClassCheckbox" label="Custom Class Checkbox" value="" className="test-class" />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders a pre-selected Checkbox correctly', () => {
        const tree = renderer
            .create(<Checkbox id="customClassCheckbox" label="Custom Class Checkbox" value={true} className="test-class" />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});