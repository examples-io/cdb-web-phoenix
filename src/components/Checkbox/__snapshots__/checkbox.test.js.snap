// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox tests renders Checkbox correctly 1`] = `
<label
  className="form__input form__input--inline form__input--checkbox "
  htmlFor="testCheckbox"
>
  <input
    checked={false}
    className="input input--checkbox"
    disabled={false}
    id="testCheckbox"
    onChange={undefined}
    type="checkbox"
    value=""
  />
  <span
    className="checkbox"
  />
  <div
    className="label label--inline label--checkbox"
  >
    Test Checkbox
  </div>
</label>
`;

exports[`Checkbox tests renders Checkbox error correctly 1`] = `
<label
  className="form__input form__input--inline form__input--checkbox "
  htmlFor="errorCheckbox"
>
  <input
    checked={false}
    className="input input--checkbox input--error"
    disabled={false}
    id="errorCheckbox"
    onChange={undefined}
    type="checkbox"
    value=""
  />
  <span
    className="checkbox"
  />
  <div
    className="label label--inline label--checkbox label--error"
  >
    Error Checkbox
  </div>
</label>
`;

exports[`Checkbox tests renders Checkbox with custom class correctly 1`] = `
<label
  className="form__input form__input--inline form__input--checkbox test-class"
  htmlFor="customClassCheckbox"
>
  <input
    checked={false}
    className="input input--checkbox"
    disabled={false}
    id="customClassCheckbox"
    onChange={undefined}
    type="checkbox"
    value=""
  />
  <span
    className="checkbox"
  />
  <div
    className="label label--inline label--checkbox"
  >
    Custom Class Checkbox
  </div>
</label>
`;

exports[`Checkbox tests renders a pre-selected Checkbox correctly 1`] = `
<label
  className="form__input form__input--inline form__input--checkbox test-class"
  htmlFor="customClassCheckbox"
>
  <input
    checked="checked"
    className="input input--checkbox"
    disabled={false}
    id="customClassCheckbox"
    onChange={undefined}
    type="checkbox"
    value={true}
  />
  <span
    className="checkbox"
  />
  <div
    className="label label--inline label--checkbox"
  >
    Custom Class Checkbox
  </div>
</label>
`;

exports[`Checkbox tests renders disabled Checkbox correctly 1`] = `
<label
  className="form__input form__input--inline form__input--checkbox "
  htmlFor="disabledCheckbox"
>
  <input
    checked={false}
    className="input input--checkbox"
    disabled={true}
    id="disabledCheckbox"
    onChange={undefined}
    type="checkbox"
    value=""
  />
  <span
    className="checkbox"
  />
  <div
    className="label label--inline label--checkbox"
  >
    Disabled Checkbox
  </div>
</label>
`;
